src/circom/r1cs_reader.rs

```diff
@@ -41,7 +44,7 @@ pub struct R1CSFile<E: PairingEngine> {
 }
 
 impl<E: PairingEngine> R1CSFile<E> {
-    pub fn new<R: Read>(mut reader: R) -> Result<R1CSFile<E>> {
+    pub fn new<R: Read + Seek>(mut reader: R) -> Result<R1CSFile<E>> {
```

Comment:
Given the new type restriction, do you mind adding a comment on how to convert a Read type to also be Seek? Suspect just giving an example of Cursor would be good enough

Reply:
I think this will be helpful. <https://github.com/poma/zkutil/pull/7/files#diff-1c228263247f6272bf2423aecf5e5f659705f35599259f1e0d79e2d036e7daa8R202>

Reply:
yeah just adding this as a nice doctest ```rust,ignore``` would be good

Reply:
Something like this? https://github.com/gakonst/ark-circom/pull/13/commits/12bca6140392245a7564629eace6c4c2b06a6e01